tests/installation/data_integrity.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify data integrity of the images provided by comparing checksums.
+# Maintainer: Joaquín Rivera <jeriveramoya@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use Digest::file 'digest_file_hex';
+
+sub run {
+    diag "Comparing data integrity calculated with SHA256 digest against checksum from IBS/OBS via rsync.pl";
+    foreach my $image (grep { /^CHECKSUM_/ } keys %bmwqemu::vars) {
+        my $checksum = get_required_var $image;
+        $image =~ s/CHECKSUM_//;
+        my $image_name = get_required_var $image;
+        my $digest = digest_file_hex($image_name, "SHA-256");
```

Comment:
I'm a bit surprised that it works, maybe you can help me to understand it. So iso and hdd images are stored in different directories on the worker. How does digest_file_hex finds them in current working directory just by the name of a file? Could you please explain me this part?

Reply:
vars.json contains the full path. Variable renamed to make it more explicit.

Reply:
Ah, you're right. It's shown differently on the settings page. Thanks!